primitives/teerex/src/lib.rs

```diff
@@ -92,6 +93,94 @@ pub enum SgxStatus {
 	ConfigurationNeeded,
 }
 
+pub type OpaqueSigner = BoundedVec<u8, ConstU32<66>>;
+pub type EnclaveFingerprint = H256;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
+pub enum AnySigner {
+	Opaque(OpaqueSigner),
+	Known(MultiSigner),
+}
+
+impl From<MultiSigner> for AnySigner {
+	fn from(signer: MultiSigner) -> Self {
+		AnySigner::Known(signer)
+	}
+}
```

Comment:
We could remove these manual implementations by importing `derive_more` and then derive `From`

Reply:
much wow! love it!